tcti/tcti.c

```diff
@@ -96,3 +100,31 @@ TSS2_RC tcti_receive_checks (
 
     return TSS2_RC_SUCCESS;
 }
+
+ssize_t write_all (
+    int fd,
+    const uint8_t *buf,
+    size_t size)
+{
+    ssize_t written = 0;
+    size_t written_total = 0;
+
+    do {
+        LOG_DEBUG("writing %zu bytes starting at 0x%" PRIxPTR " to fd %d",
+                  size - written_total,
+                  (uintptr_t)buf + written_total,
+                  fd);
+        written = TEMP_RETRY (write (fd,
+                                     (const char*)&buf [written_total],
+                                     size - written_total));
+        if (written >= 0) {
+            LOG_DEBUG ("wrote %zd bytes to fd %d", written, fd);
+            written_total += (size_t)written;
+        } else {
+            LOG_ERROR ("failed to write to fd %d: %s", fd, strerror (errno));
+            return written_total;
```

Comment:
return written_total ? written_total : -1;
According to man 3 write 0 is not a valid return value from write(). It should be either number of bytes written (if any) or -1.

Reply:
I think you want `man 2 write`. Linux allows for a 0 return from `write` when nothing is written.